### PR TITLE
[WIP]feat: implement change password (#32)

### PR DESCRIPTION
**Pull request by ashishsurana**
**Type:**

The following has been addressed in the PR:

*  There is a related issue?
*  Unit or Functional tests are included in the PR

**Description:**

### CODE REVIEW

**Comment by ayusharma** [comment]
No tests ??

File: src/htpasswd.js [hunk 1]
```diff
@@ -204,4 +205,61 @@ export default class HTPasswd {
...
   }
+
+  /**
+   * changePassword - change password for existing user.
+   * @param {string} user
+   * @param {string} password
+   * @param {function} cd
+   * @returns {function}
+   */
+  changePassword(
```
Comment (ayusharma) [inline src/htpasswd.js]: change `function` name.

**Comment by ayusharma** [comment]
change commit message to type `feat`.. see PR name

File: src/utils.js [hunk 2]
```diff
@@ -148,3 +148,52 @@ export function sanityCheck(
...
+  user: string,
+  passwd: string,
+  newPasswd: string
+): string {
+  if (crypt3) {
+    passwd = crypt3(passwd);
+    newPasswd = crypt3(newPasswd);
+  } else {
+    passwd =
+      '{SHA}' +
```
Comment (juanpicado) [inline src/utils.js]: https://developers.google.com/web/updates/2015/01/ES6-Template-Strings

File: src/utils.js [hunk 3]
```diff
@@ -148,3 +148,52 @@ export function sanityCheck(
...
+      crypto
+        .createHash('sha1')
+        .update(newPasswd, 'binary')
+        .digest('base64');
+  }
+
+  let lines = body.split('\n');
+  lines = lines.map(line => {
+    let args = line.split(':', 3);
```
Comment (juanpicado) [inline src/utils.js]: I'd suggest a better name here. `args`

Comment (juanpicado) [reply src/utils.js]: If we expect always 3. We can destructurate 
```
 let [one, two, three] = line.split(':', 3)
``

**Comment by ashishsurana** [comment]
@juanpicado @ayusharma Please review the above changes

File: src/___tests___/htpassed.test.js [hunk 4]
```diff
@@ -76,4 +76,35 @@ describe('HTPasswd', () => {
     };
     wrapper.adduser('usernotpresent', 'somerandompassword', callback);
   });
+
+  it('changePassword - it should throw an error', done => {
+    const callback = (a, b) => {
```
Comment (juanpicado) [inline src/___tests___/htpassed.test.js]: `a, b` is not a good name, I'd chose something like 3 months later reminder what means `a` or `b`.

Comment (juanpicado) [reply src/___tests___/htpassed.test.js]: The same below this file

File: src/___tests___/htpassed.test.js [hunk 5]
```diff
@@ -76,4 +76,35 @@ describe('HTPasswd', () => {
...
+    };
+    wrapper.changePassword(
+      'usernotpresent',
+      'oldPassword',
+      'newPassword',
+      callback
+    );
+  });
+
+  it('changePassword - it should change passwor', done => {
```
Comment (juanpicado) [inline src/___tests___/htpassed.test.js]: `passwor` typo

**Comment by ayusharma** [comment]
@ashishsurana Please think like this:

ChangePasswordMethod
1. authenticate user
2. write new user credentials. 

Connect me on discord. I'll explain you in more detail.

**Comment by juanpicado** [comment]
@ashishsurana @ayusharma I'll take care of this.

**Pull request closed without merging**